### ISSUE TITLE
Deal with categorical features

### ISSUE THREAD

**Issue by stephanegaiffas** [pyensemble/wildwood#2]
We need to deal properly with categorical features. For now, only for the binary classification case (dealing with K class classification will require to grow K trees instead of one, and use a one-versus all strategy, see #3).
For this we need :

- Handle the `categorical_features` argument used in `ForestBinaryClassifier`. For now we can assume that the user MUST specify which features are categorical. If the input is a pandas DataFrame, we should look at the dtype and look which columns have the "category" dtype. Maybe it's handled already by the `Binner` we use from `sklearn`. We need to save this information correctly in the `tree_context` of somewhere central that can be access by all trees
- In the node_context, we compute in the `compute_node_context` the `y_sum` which contains, for each feature and each bin, the sum of the labels per class. Then this is used in the `_split.find_best_split_along_feature` in the `y_sum_in_bins = node_context.y_sum[f]`. If the feature is categorical, we can simply sort `y_sum_in_bins[:, 0]` (the sum wrt the first label class) and use this ordering in the `for bin in range(n_bins):`. This will work for binary classification and for regression as well,
- We can look at https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/ensemble/_hist_gradient_boosting/splitting.pyx#L781 for inspiration
